tests/compiler/test_warnings.py

```diff
@@ -151,7 +152,7 @@ def test_1918_shadow_warning_for_loop(snippetcompiler):
         assert str(w1.message) == message % (2, 4)
 
 
-def test_deprecation_warning_nullable(snippetcompiler):
+def test_deprecation_exception_nullable(snippetcompiler):
```

Comment:
The contents of this test look good to me but I would rename it (it doesn't have anything to do with deprecation anymore) and move it to `tests/compiler/test_null.py` (it doesn't have anything to do with warnings anymore).